Unindexed qualifier throws NPE when ArcProcessor validates beans

### Describe the bug

Don't know if this expected but the experience may be improved as it is the second time I lost some time finding out what was going on.

I was trying to register two synthetic beans `SqsClientBuilder` and `SqsClient` . The second one having an injection point of the first one with a custom qualifier. Both beans have a custom qualifier. Those qualifiers were not register as per documented https://quarkus.io/guides/cdi-integration#use-case-my-annotation-is-not-recognized-as-a-qualifier-or-an-interceptor-binding.

I got an NPE because `requiredClazz` is null here for the first qualifier.
https://github.com/quarkusio/quarkus/blob/542f5530b336ddd75a22e84fa878547a053f492d/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java#L684

```
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: java.lang.NullPointerException: Cannot invoke "org.jboss.jandex.ClassInfo.method(String, org.jboss.jandex.Type[])" because "requiredClazz" is null
        at io.quarkus.arc.processor.Beans.hasQualifier(Beans.java:684)
```

### Expected behavior

No NPE and as much as possible, a meaningfull error message.

### Actual behavior

Inspecting `beanDeployment.beanArchiveImmutableIndex`, it contains only the qualifier of the second bean, so something seems to have added it automatically (maybe because it is used in the app project), but not the required one.
If `hasQualifier` return false when `requiredClazz` is null, then, I have an more descriptive error but it is still meaningless.

```
Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type software.amazon.awssdk.services.sqs.SqsClientBuilder and qualifiers [@io.quarkus.amazon.common.AmazonClientBuilder("test")]
        - synthetic injection point
        - declared on SYNTHETIC bean [types=[software.amazon.awssdk.services.sqs.SqsClient, java.lang.Object], qualifiers=[@Any, @io.quarkus.amazon.common.AmazonClient("test")], target=n/a]
        The following beans match by type, but none has matching qualifiers:
        - SYNTHETIC bean [types=[java.lang.Object, software.amazon.awssdk.services.sqs.SqsClientBuilder], qualifiers=[@io.quarkus.amazon.common.AmazonClientBuilder("test"), @Any], target=n/a]
        - SYNTHETIC bean [types=[java.lang.Object, software.amazon.awssdk.services.sqs.SqsClientBuilder], qualifiers=[@Default, @Any], target=n/a]
```

I'm wondering why `io.quarkus.amazon.common.AmazonClient` is a known qualifier and `io.quarkus.amazon.common.AmazonClientBuilder` is not.
Why ArcProcessor knows about the qualifier as the error message shows it and we still need to register the annotation manually for AdditionalBeanBuildItem.
Could it be possible to add a message when `requiredClazz` is null that tells that a qualifier should be registered manually ?

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

3.11

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java
@@ -680,6 +680,12 @@ public final class Beans {
                     //as this is called in a tight loop we only do it if necessary
                     values = new ArrayList<>();
                     Set<String> nonBindingFields = beanDeployment.getQualifierNonbindingMembers(requiredQualifier.name());
+                    if (requiredClazz == null) {
+                        throw new IllegalStateException("Failed to find bean qualifier class with name "
+                                + requiredQualifier.name() + " in application index. Make sure the class is part of "
+                                + "the Jandex index. Classes that are not subject to discovery can be registered via "
+                                + "AdditionalBeanBuildItem and non-qualifier annotations can use QualifierRegistrarBuildItem");
+                    }
                     for (AnnotationValue val : requiredQualifier.valuesWithDefaults(beanDeployment.getBeanArchiveIndex())) {
                         if (!requiredClazz.method(val.name()).hasAnnotation(DotNames.NONBINDING)
                                 && !nonBindingFields.contains(val.name())) {

